test/jdk/java/text/Format/DecimalFormat/LargeExponentsTest.java

```diff
@@ -0,0 +1,152 @@
+        List<Arguments> args = new ArrayList<>();
+        // Start with a base value that should be parsed as infinity
+        String baseValue = "12234.123E1100";
+        // Continuously add to the String until we trigger the overflow condition
+        for (int i = 0; i < 100; i++) {
+            StringBuilder bldr = new StringBuilder();
+            // Add to exponent
+            bldr.append(baseValue).append("1".repeat(i));
+            // Add to mantissa
+            bldr.insert(0, "1".repeat(i));
+            args.add(Arguments.of(
+                    // Prepend "-" to exponent if negative
+                    negative ? bldr.insert(bldr.indexOf("E")+1, "-").toString() : bldr.toString(),
+                    // Expect 0 if negative, else infinity
+                    negative ? 0.0 : Double.POSITIVE_INFINITY));
+        }
+        return args;
+    }
+
+    // The provided values are all from the JBS issue
+    // These contain exponents that exceed Integer.MAX_VALUE, but not Long.MAX_VALUE
+    private static Stream<Arguments> bugReportValues() {
+        return Stream.of(
+                Arguments.of("0.123E1", 1.23),
+                Arguments.of("0.123E309", 1.23E308),
+                Arguments.of("0.123E310", Double.POSITIVE_INFINITY),
+                Arguments.of("0.123E2147483647", Double.POSITIVE_INFINITY),
+                Arguments.of("0.123E2147483648", Double.POSITIVE_INFINITY),
+                Arguments.of("0.0123E2147483648", Double.POSITIVE_INFINITY),
+                Arguments.of("0.0123E2147483649", Double.POSITIVE_INFINITY),
+                Arguments.of("1.23E2147483646", Double.POSITIVE_INFINITY),
+                Arguments.of("1.23E2147483647", Double.POSITIVE_INFINITY),
+                Arguments.of("0.123E4294967296", Double.POSITIVE_INFINITY),
+                Arguments.of("0.123E-322", 9.9E-324),
+                Arguments.of("0.123E-323", 0.0),
+                Arguments.of("0.123E-2147483648", 0.0),
+                Arguments.of("0.123E-2147483649", 0.0)
+        );
+    }
+
+    // Some odd edge case values to ensure parse correctness
+    private static Stream<Arguments> edgeCases() {
+        return Stream.of(
+                // 0 exponent
+                Arguments.of("1.23E0", 1.23),
+                // Leading zeroes
+                Arguments.of("1.23E0000123", 1.23E123),
+                // Leading zeroes - Past Long.MAX_VALUE length
+                Arguments.of("1.23E00000000000000000000000000000000000000000123", 1.23E123),
+                // Trailing zeroes
+                Arguments.of("1.23E100", 1.23E100),
+                // Long.MAX_VALUE length
+                Arguments.of("1.23E1234567891234567800", Double.POSITIVE_INFINITY),
+                // Long.MAX_VALUE with trailing zeroes
+                Arguments.of("1.23E9223372036854775807000", Double.POSITIVE_INFINITY),
+                // Long.MIN_VALUE
+                Arguments.of("1.23E-9223372036854775808", 0.0)
+        );
```

Comment:
I would suggest adding one more test case to the edge cases:
```java
Arguments.of("0.0123E-2147483648", 0.0)
```
This will test the adjustment of the `digits.decimalAt` field for an exponent that is within the range of integer, but due to the mantissa not being in its standardized form an overflow will occure non the less.

Reply:
Right, that's a good test case to have, (which exposed a needed update in the implementation).